datahub-web-react/src/app/shared/tags/TagGroup.tsx

```diff
@@ -0,0 +1,113 @@
+import { Modal, Tag } from 'antd';
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import styled from 'styled-components';
+import { FetchResult } from '@apollo/client';
+
+import { useEntityRegistry } from '../../useEntityRegistry';
+import { EntityType, GlobalTags, GlobalTagsUpdate } from '../../../types.generated';
+import { UpdateDatasetMutation } from '../../../graphql/dataset.generated';
+import { convertTagsForUpdate } from './utils/convertTagsForUpdate';
+import AddTagModal from './AddTagModal';
+
+type Props = {
+    uneditableTags?: GlobalTags | null;
+    editableTags?: GlobalTags | null;
+    canRemove?: boolean;
+    canAdd?: boolean;
+    updateTags?: (
+        update: GlobalTagsUpdate,
+    ) => Promise<FetchResult<UpdateDatasetMutation, Record<string, any>, Record<string, any>>>;
+    onOpenModal?: () => void;
+    maxShow?: number;
+};
+
+const AddNewTag = styled(Tag)`
+    cursor: pointer;
+`;
+
+export default function TagGroup({
+    uneditableTags,
+    editableTags,
+    canRemove,
+    canAdd,
+    updateTags,
+    onOpenModal,
+    maxShow,
+}: Props) {
+    const entityRegistry = useEntityRegistry();
+    const [showAddModal, setShowAddModal] = useState(false);
+
+    const removeTag = (urnToRemove: string) => {
+        onOpenModal?.();
+        const tagToRemove = editableTags?.tags?.find((tag) => tag.tag.urn === urnToRemove);
+        const newTags = editableTags?.tags?.filter((tag) => tag.tag.urn !== urnToRemove);
+        Modal.confirm({
+            title: `Do you want to remove ${tagToRemove?.tag.name} tag?`,
+            content: `Are you sure you want to remove the ${tagToRemove?.tag.name} tag?`,
+            onOk() {
+                updateTags?.({ tags: convertTagsForUpdate(newTags || []) });
+            },
+            onCancel() {},
+            okText: 'Yes',
+            maskClosable: true,
+            closable: true,
+        });
+    };
+
+    let renderedTags = 0;
+
+    return (
+        <div>
+            {/* uneditable tags are provided by ingestion pipelines exclusively */}
+            {uneditableTags?.tags?.map((tag) => {
+                renderedTags += 1;
+                if (maxShow && renderedTags > maxShow) return null;
+                return (
+                    <Link to={`/${entityRegistry.getPathName(EntityType.Tag)}/${tag.tag.urn}`} key={tag.tag.urn}>
+                        <Tag color="blue" closable={false}>
+                            {tag.tag.name}
+                        </Tag>
+                    </Link>
+                );
+            })}
+            {/* editable tags may be provided by ingestion pipelines or the UI */}
+            {editableTags?.tags?.map((tag) => {
```

Comment:
Seems there is some duplicated logic here -- maybe in the future we can consider wrapping in a fcn

Reply:
+1